pkg/module/sync/runner_impl.go

```diff
@@ -66,134 +64,119 @@ func (r *RunnerImpl) Sync(ctx context.Context, kyma *v1alpha1.Kyma,
 	if errs != nil {
 		return fmt.Errorf("ServerSideApply failed (after %s): %w", ssaFinish, types.NewMultiError(errs))
 	}
-	baseLogger.V(int(zap.DebugLevel)).Info("ServerSideApply finished", "time", ssaFinish)
+	baseLogger.V(log.DebugLevel).Info("ServerSideApply finished", "time", ssaFinish)
 	return nil
 }
 
-func (r *RunnerImpl) getModule(ctx context.Context, module *manifestV1alpha1.Manifest) error {
+func (r *RunnerImpl) getModule(ctx context.Context, module client.Object) error {
 	return r.Get(ctx, client.ObjectKey{Namespace: module.GetNamespace(), Name: module.GetName()}, module)
 }
 
-func (r *RunnerImpl) updateModule(ctx context.Context, name string, kyma *v1alpha1.Kyma,
+func (r *RunnerImpl) updateModule(ctx context.Context, kyma *v1alpha1.Kyma,
 	module *common.Module,
 ) error {
-	if err := r.setupModule(module, kyma, name); err != nil {
+	if err := r.setupModule(module, kyma); err != nil {
 		return err
 	}
-	obj, err := r.converter.ConvertToVersion(module.Manifest, r.versioner)
+	obj, err := r.converter.ConvertToVersion(module.Object, r.versioner)
 	if err != nil {
 		return err
 	}
 	clObj := obj.(client.Object)
-	if err := r.Patch(ctx, obj.(client.Object),
+	if err := r.Patch(ctx, clObj,
 		client.Apply,
 		client.FieldOwner(kyma.Labels[v1alpha1.ManagedBy]),
 		client.ForceOwnership,
 	); err != nil {
-		return fmt.Errorf("error patching custom resource of type %s %w", name, err)
+		return fmt.Errorf("error applying manifest %s: %w", client.ObjectKeyFromObject(module), err)
 	}
-	module.UpdateStatusAndReferencesFromUnstructured(clObj.(*manifestV1alpha1.Manifest))
+	module.Object = clObj
 
 	return nil
 }
 
-func (r *RunnerImpl) setupModule(module *common.Module, kyma *v1alpha1.Kyma, name string) error {
+func (r *RunnerImpl) setupModule(module *common.Module, kyma *v1alpha1.Kyma) error {
 	// set labels
-	module.ApplyLabels(kyma, name)
+	module.ApplyLabelsAndAnnotations(kyma)
 
 	if module.GetOwnerReferences() == nil {
 		// set owner reference
-		if err := controllerutil.SetControllerReference(kyma, module.Manifest, r.Scheme()); err != nil {
+		if err := controllerutil.SetControllerReference(kyma, module.Object, r.Scheme()); err != nil {
 			return fmt.Errorf("error setting owner reference on component CR of type: %s for resource %s %w",
-				name, kyma.Name, err)
+				module.GetName(), kyma.Name, err)
 		}
 	}
 
 	return nil
 }
 
 func (r *RunnerImpl) SyncModuleStatus(ctx context.Context, kyma *v1alpha1.Kyma, modules common.Modules) bool {
```

Comment:
since currently we are not using the returned value to determine if need to do status update, we can simplify this logic without the return value.